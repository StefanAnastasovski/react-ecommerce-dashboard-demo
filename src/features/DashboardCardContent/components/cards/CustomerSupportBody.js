import React from "react";

import { Box } from "@mui/material";

import {
  LINK_N_TEXT_FONT_SIZE,
} from "../../../../data/constants";
import AGENT_PIC from "../../../../assets/images/agent-profile-pic.jpeg";

import { DASHBOARD_API } from "../../../../data/dashboardConfig";

import Paragraph from "../../../../components/Typography/Paragraph";
import Image from "../../../../components/Image";

const CustomerSupportBody = () => {
  const {
    cs_agent: { name },
  } = DASHBOARD_API;

  const styledImage = {
    sx: { width: "100%", height: "100%", borderRadius: "50%" },
  };
  const agentImageDimensions = {
    sx: {
      width: "40px",
      height: "40px",
    },
  };

  return (
    <Box sx={{ display: "flex", alignItems: "center" }} my={2}>
      <Box {...agentImageDimensions}>
        <Image
          source={AGENT_PIC}
          alt={`Agent ${name} profile pic`}
          {...styledImage}
        />
      </Box>
      <Paragraph fontSize={LINK_N_TEXT_FONT_SIZE} my={3} ml={2.75}>
        {name} is here to help you.
      </Paragraph>
    </Box>
  );
};

export default CustomerSupportBody;
