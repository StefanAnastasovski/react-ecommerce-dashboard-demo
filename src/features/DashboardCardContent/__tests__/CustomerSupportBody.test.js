import React from "react";
import { shallow } from "enzyme";

import CustomerSupportBody from "../components/cards/CustomerSupportBody";

describe("<CustomerSupportBody />", () => {
  describe("render()", () => {
    it("Shallow rendering - MatchSnapshot", () => {
      const component = shallow(<CustomerSupportBody />);
      expect(component).toMatchSnapshot();
    });
  });
});
