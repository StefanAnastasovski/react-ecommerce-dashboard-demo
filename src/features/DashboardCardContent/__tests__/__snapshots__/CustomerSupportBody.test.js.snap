// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerSupportBody /> render() Shallow rendering - MatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomerSupportBody />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(Box)
          sx={
            Object {
              "height": "40px",
              "width": "40px",
            }
          }
        >
          <Image
            alt="Agent Natalie profile pic"
            source="agent-profile-pic.jpeg"
            styleProps={
              Object {
                "borderRadius": "50%",
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </ForwardRef(Box)>,
        <Paragraph
          styleProps={
            Object {
              "fontSize": "17px",
              "margin": "1em 0",
              "marginLeft": "1em",
            }
          }
        >
          Natalie
           is here to help you.
        </Paragraph>,
      ],
      "my": 2,
      "sx": Object {
        "alignItems": "center",
        "display": "flex",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Image
            alt="Agent Natalie profile pic"
            source="agent-profile-pic.jpeg"
            styleProps={
              Object {
                "borderRadius": "50%",
                "height": "100%",
                "width": "100%",
              }
            }
          />,
          "sx": Object {
            "height": "40px",
            "width": "40px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Agent Natalie profile pic",
            "source": "agent-profile-pic.jpeg",
            "styleProps": Object {
              "borderRadius": "50%",
              "height": "100%",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "children": [Function],
            "component": [Function],
            "sx": [Function],
          },
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Natalie",
            " is here to help you.",
          ],
          "styleProps": Object {
            "fontSize": "17px",
            "margin": "1em 0",
            "marginLeft": "1em",
          },
        },
        "ref": null,
        "rendered": Array [
          "Natalie",
          " is here to help you.",
        ],
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "children": [Function],
        "component": [Function],
        "sx": [Function],
      },
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(Box)
            sx={
              Object {
                "height": "40px",
                "width": "40px",
              }
            }
          >
            <Image
              alt="Agent Natalie profile pic"
              source="agent-profile-pic.jpeg"
              styleProps={
                Object {
                  "borderRadius": "50%",
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </ForwardRef(Box)>,
          <Paragraph
            styleProps={
              Object {
                "fontSize": "17px",
                "margin": "1em 0",
                "marginLeft": "1em",
              }
            }
          >
            Natalie
             is here to help you.
          </Paragraph>,
        ],
        "my": 2,
        "sx": Object {
          "alignItems": "center",
          "display": "flex",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Image
              alt="Agent Natalie profile pic"
              source="agent-profile-pic.jpeg"
              styleProps={
                Object {
                  "borderRadius": "50%",
                  "height": "100%",
                  "width": "100%",
                }
              }
            />,
            "sx": Object {
              "height": "40px",
              "width": "40px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Agent Natalie profile pic",
              "source": "agent-profile-pic.jpeg",
              "styleProps": Object {
                "borderRadius": "50%",
                "height": "100%",
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "children": [Function],
              "component": [Function],
              "sx": [Function],
            },
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Natalie",
              " is here to help you.",
            ],
            "styleProps": Object {
              "fontSize": "17px",
              "margin": "1em 0",
              "marginLeft": "1em",
            },
          },
          "ref": null,
          "rendered": Array [
            "Natalie",
            " is here to help you.",
          ],
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "children": [Function],
          "component": [Function],
          "sx": [Function],
        },
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
